analyzers/src/SonarAnalyzer.CSharp/Wrappers/ObjectCreationFactory.cs

```diff
@@ -95,8 +95,9 @@ private sealed class ImplicitObjectCreation : IObjectCreation
             public bool IsKnownType(KnownType knownType, SemanticModel semanticModel) =>
                 semanticModel.GetTypeInfo(objectCreation).Type.Is(knownType);
 
+            // Return null if TypeSymbol returns null to avoid AD0001 due to this issue: https://github.com/dotnet/roslyn/issues/70041
             public string TypeAsString(SemanticModel semanticModel) =>
-                TypeSymbol(semanticModel).Name;
+                TypeSymbol(semanticModel)?.Name;
```

Comment:
This is used in two places:  `CSharpUsageCollector` inside a HashSet.Contains (this should be fine) and in `ClearTextProtocolsAreSensitive` in a `Regex.IsMatch`. `IsMatch` throws if it's called with null. See: https://learn.microsoft.com/en-us/dotnet/api/system.text.regularexpressions.regex.ismatch?view=net-7.0#system-text-regularexpressions-regex-ismatch(system-string)